File: conversion.py
1 + 1

print(5)

a = 5

dictionary = {
  'hello': 'hi',
  'there': 'that'
}

word = dictionary['hello']

b = 'hello'

b[0] = 'j'

array = ['s', 5, 8]

third = array[2]

var zero_to_three = array.slice(0, 3)

last_element = array[len(array) - 1]

new_array = []

new_array.append(1)
new_array.append(2)
new_array.pop()
new_array.insert(0, 'a')
new_array.pop(0)


#############################################


def square(num):
  return num * num

c = square(a)

if True:
  return True
elif False:
  return None
elif False or True:
  return True and False
elif 0 == 1:
  return 0
else:
  return 'error'

while a > 0:
  a -= 1

for x in range(5):
  print(x ** 2)


#############################################


sheep = lambda x: x

def higher_order(num):
  def inception():
    nonlocal num
    num += 5
  inception()
  return num

higher_order(6) # returns 11

mess_this = 1
def will_do():
  global mess_this
  mess_this += 6

will_do()

print(mess_this) # 7

File: solutions/conversion.js
1 + 1

console.log(5);

var a = 5;

var dictionary = {
  'hello': 'hi',
  'there': 'that'
};

var dictionary = {
  hello: 'hi',
  there: 'that'
}; 

word = dictionary['hello'];
word = dictionary.hello;

var b = 'hello';

b[0] = 'j'; // b = 'jello'

var array = ['s', 5, 8];

var third = array[2];

var zeroToThree = array.slice(0,3);

var lastElement = array[array.length - 1];

var newArray = [];

newArray.push(1);
newArray.push(2);
newArray.pop(); // 2
newArray.unshift('a');
newArray.shift(); // 'a'


/////////////////////////////////////////////


var square = function(num) {
  return num * num;
};

// I don't use this ever:
function square(num) {
  return num * num;
}

c = square(a);

if (true) {
  return true;
}
else if (false) {
  return null;
  return undefined;
}
else if (false || true) {
  return true && false;
}
else if (0 === 1) {
  return 0;
}
else {
  return 'error';
}

while (a > 0) {
  a -= 1;
}

for (var x = 0; x < 5; x += 1) {
  console.log(Math.pow(x, 2));
}


/////////////////////////////////////////////


var sheep = function(x) { return x; };

var higherOrder = function(num) {
  var inception = function() {
    num += 5;
  };
  inception();
  return num;
};

higherOrder(6); // returns 11

var messThis = 1;
var willDo = function() {
    messThis += 6;
};

willDo();

console.log(messThis); // 7
